test-suite/shared/src/test/require-jdk8/org/scalajs/testsuite/javalib/time/TemporalAccessorTest.scala

```diff
@@ -7,8 +7,8 @@ import org.junit.Test
 import org.junit.Assert._
 import org.scalajs.testsuite.utils.AssertThrows._
 
-abstract class TemporalAccessorTest {
-  val samples: Seq[TemporalAccessor]
+abstract class TemporalAccessorTest[A <: TemporalAccessor] {
+  val samples: Seq[A]
```

Comment:
Yeah, thought the same myself... I didn't really like to add a type parameter to the test class but the reason for that `A` was driven more by the new:

```scala
def expectedRangeFor(accessor: A, field: TemporalField): ValueRange = field.range()
```

The _expected range_ for a particular `TemporalField` depends not only on the field, but in the actual sample instance being tested at a particular point. And then the implementation of the _special cases_ depends on fields/methods declared on the `TemporalAccessor` concrete implementation.

This is a snippet from `MonthDayTest`:

```scala
override def expectedRangeFor(accessor: MonthDay, field: TemporalField): ValueRange = {
    field match {
      case DAY_OF_MONTH =>
        ValueRange.of(1, accessor.getMonth.minLength(), accessor.getMonth.maxLength())

      case _ =>
        super.expectedRangeFor(accessor, field)
    }
  }
```

This is a snippet from `LocalDateTest`:

```scala
override def expectedRangeFor(accessor: LocalDate, field: TemporalField): ValueRange = {
    field match {
      case DAY_OF_MONTH => ValueRange.of(1, accessor.lengthOfMonth)
      case DAY_OF_YEAR  => ValueRange.of(1, accessor.lengthOfYear)

      case ALIGNED_WEEK_OF_MONTH =>
        ValueRange.of(1, if (accessor.lengthOfMonth > 28) 5 else 4)

      case YEAR_OF_ERA =>
        val maxYear = if (accessor.getEra == IsoEra.CE) 999999999 else 1000000000
        ValueRange.of(1, maxYear)

      case _ =>
        super.expectedRangeFor(accessor, field)
    }
  }
```

None of the methods `MonthDay.getMonth`, `LocalDate.lengthOfMonth`, `LocalDate.lengthOfYear` are accessible from the `TemporalAccessor` type as they are implemented by the different concrete classes. So I ended up adding the type parameter to basically avoid an `accessor.asInstanceOf[MonthDay]` cast in the concrete test classes.